### ISSUE TITLE
Apple Services after not available after some time - reconnecting to the network will solve it

### ISSUE THREAD

**Issue by pictosun** [AdguardTeam/AdGuardDNS#777]
### Platform

macOS

### Protocol

DNS-over-HTTPS

### Do you use AdGuard app?

No I don't

### Your configuration

- macOS/iOS (newest system update)
- using .mobileconfig Profiles on macOS/iOS (and also tried AdGuard Pro App with AdGuard DNS as native profile)
- tried DoT/DoH with .mobileconfig and DoQ (with a 3rd party app and without .mobileconfig)
- block Apple private relay function is 'not' used (and I do have apple services on my whitelist)

### Traceroute to AdGuard DNS

./. doesn't matter

### Issue Details

- While using iOS/macOS devices after several amount of time (especially when changing networks from mobile to WIFI and so on) it does happen that Apple services like App Store, Apple Music do not work anymore.
- Even force quitting the apps don't bring back connection to the services
- when putting the devices into flight mode and back online the services do work again

### Expected Behavior

It should work all the time

### Actual Behavior

When using AdGuard DNS (with some blocklists) the problem exists and reappears often.

### Additional Information

When using some other DNS services like NextDNS, ControlD etc. (nearly same blocklists enabled) the issue is not existent

It looks like it's something correlated to AdGuard DNS service at all?

**Comment by Chinaski1** [comment]
Hello there!

Please clarify the following information:
- Do you use third-party software like a VPN?
- Which blocklists are enabled?
- What [DNS server]( https://dns.adguard-dns.com/info.txt) are you connected to?

**Comment by pictosun** [comment]
@Chinaski1 I can give you the update, that the issue just appeared again when changing WAN networks. 

I also checked the query logs and cannot see any block within this time and/or domains... So the blocklists shouldn't be the issue and it looks like it is something else.

Changing DNS profile to some other provider solves the issue. Going back to AdGuard DNS after that also works.

**Comment by pictosun** [comment]
Just do give you an update over here... When changing to an other DNS service (Contr...D, Ne...DNS) I do not have those issues (using nearly the same blocklists) and I think it's not related to blocklists itself, as there is nothing showing up within blocked queries.

So something must be different within AdGuard DNS?
Could it maybe something correlated to this: https://github.com/AdguardTeam/AdGuardDNS/issues/227

**Comment by hagezi** [comment]
I use AdGuard DNS via mobileconfig on 10 Apple devices, including iPhones, iPads, Macbooks and Apple TV and do not have this problem and cannot reproduce it.

**Comment by pictosun** [comment]
Thanks for your feedback. Really a strange behavior. For me I can more or less reproduce it on several devices with several kind of blocklists (don’t think the lists are the issue).

What I (maybe) found out is, that it is eventually related to changing WAN networks with IPv4/IPv6 changing.
At home I do have dual stack, at work IPv4 and mobile it depends on.

Don’t know if this is maybe the case and has something to do with my issues.

As it happended to several macOS/iOS versions I don’t think that this is the case. Especially while hagezi says that everything is fine for his devices. 

Don’t know how can look into more details to report it over here to find the reason why this is happening (and also only happening when using AdGuard DNS - with NextDNS and ControlD and other DNS services I don’t have those issues?!).

As I said, I will try to narrow it down so that finding the reason is easier?!

Thanks for your product and help over here.

**Comment by Batman2814** [comment]
Try unblocking this domain "mesu.apple.com"
This domain must be unblocked for the Apple TV App to work on Apple devices, but it might require other Apple services.

**Comment by pictosun** [comment]
> Try unblocking this domain "mesu.apple.com" 
mesu.apple.com is always unblocked

As mentioned above I don't think it's an issue with any blocking list at all, as it does happen regardless of entries in any list. 
So it must be something else. I think it has to do with some kind of network change (IPv4/IPv6) and maybe AdGuards way (the other players like NextDNS and ControlD,... don't do this) to check other resolvers (Google, Cloudflare) when getting a SERVFAIL response.

Or they're still doing something different within their implementation.

When trying out AdGuard Home on a VPS it also doesn't happen.

And as mentioned above it's quite hard to find out what's causing the issue.

**Comment by emeritaacuity0u** [comment]
I've been experiencing this for the last few weeks as well when accessing the Apple App Store, it will refuse to connect intermittently.

**Comment by hagezi** [comment]
> When trying out AdGuard Home on a VPS it also doesn't happen.

Are you also using AdGuard DNS (unfiltered) as upstream, if not you should test whether the problem also occurs when using it. If the problem occurs, have a look at the log and check whether one of the corresponding domains returns SERVFAIL as a result.

`https://unfiltered.adguard-dns.com/dns-query`

**Comment by pictosun** [comment]
> Are you also using AdGuard DNS (unfiltered) as upstream, if not you should test whether the problem also occurs when using it.

Thanks for your feedback. I will give it a try. 
But what about Bootstrap DNS and Private reverse DNS Servers? At the moment they're running with my Unbound resolver on the VPS. 
Do I also need to change them to test it?

**Comment by hagezi** [comment]
No, only the upstream DNS.

**Comment by pictosun** [comment]
> I've been experiencing this for the last few weeks as well when accessing the Apple App Store, it will refuse to connect intermittently.

In which scenarios for you? When changing WAN networks (home WIFI to some other networks), or don't you know?

**Comment by emeritaacuity0u** [comment]
> > I've been experiencing this for the last few weeks as well when accessing the Apple App Store, it will refuse to connect intermittently.
> 
> 
> 
> In which scenarios for you? When changing WAN networks (home WIFI to some other networks), or don't you know?

Both wifi and mobile.

DNS is setup with a configuration profile, so I can rule out AdGuard for iOS, or any other DNS app.

**Comment by pictosun** [comment]
> Are you also using AdGuard DNS (unfiltered) as upstream, if not you should test whether the problem also occurs when using it.

The problem does not occur when using it this way. But I'll keep testing it.

**Comment by hagezi** [comment]
I assume that you also use your AdGuard Home VPS via mobileconfig to at least use a similar test scenario?

**Comment by pictosun** [comment]
Yes - same devices. Made a mobileconfig (same protocol DoH) and same blocking-lists (but I don't think it's because of the blocking lists as mentioned above).

**Comment by pictosun** [comment]
Just to give a short update. Changed back (after no issues when using AGH on VPS with AdGuard Servers as Upstream) to AdGuard DNS.io mobileconfig and the issue came back immediately.

So the issue is definitively somehow within adguard-dns.io implementation.

**Comment by hagezi** [comment]
@pictosun Are you using a mobileconfig in which the home network WLAN is excluded, i.e. your local DNS is used when you are in the home network?

You could test whether the problem also occurs with a self-generated mobileconfig, you can use the following generator: https://dns.notjakob.com/tool.html

I would test an unsigned mobileconfig.

**Comment by pictosun** [comment]
> Are you using a mobileconfig in which the home network WLAN is excluded, i.e. your local DNS is used when you are in the home network?

No - I didn't exclude it. Scenario at home:

- Router: DoT profile adguard-dns.io (because of Router only supporting DoT)
- Devices: DoH mobileconfig profile (with exclusions)

At work:
- Router: no profile at all
- Devices: using same profiles as above

It is happening at work and at home when changing networks. So I don't think it's because of network exclusions or not (and other providers do work - no matter of exclusions or not)

Or did I understand you wrong?

**Comment by hagezi** [comment]
Strange, try a self-generated mobielconfig, just to rule out that this is the cause. For whatever reason.

**Comment by pictosun** [comment]
> What steps are you taking to reproduce the problem?

Normal behavior during the day. Leaving the house (and then checking Apples App Store for connection). Coming back home and so on.
Today morning I changed mobileconfig back to adguard-dns.io and after leaving the house and coming back the store was not available (no internet connection). 
When using AGH via VPS and mobileconfig those issues are not there (same with NextDNS, ControlD and others). Only adguard-dns.io with the issue.

**Comment by hagezi** [comment]
Crazy, I can't reproduce the problem.

Does the problem also occur when you switch from Home to Mobile and back to Home (WLAN off/on), or only when you switch from Home to Mobile to Work and then back to Mobile/Home?

**Comment by pictosun** [comment]
> Does the problem also occur when you switch from Home to Mobile and back to Home (WLAN off/on)

Yes. Most of the time I'm in home office so I can reproduce it quite often.

I also do use the generated mobileconfig from AdGuard. fritz.box domain is not excluded in my case. But don't know if this is the issue?!

**Comment by hagezi** [comment]
When you create a mobilconfig, there is a link to a profile constructor. You can exclude local domains so that the local queries are not made via AdGuard DNS, but directly via the locally assigned DNS. If fritz.box is excluded, *.fritz.box is resolved via the FritzBox if the clients are assigned this as upstream DNS via DHCP.

**Comment by pictosun** [comment]
@hagezi 
After some testing I can say, that excluding fritz.box doesn't solve the issue.

Today I also recognized, that it also appears when coming out of flight mode). Maybe because disabling flight mode causes a short connection to LTE/5G and after that jumps to WIFI.

So to reproduce it:
- toggle flight mode (getting mobile network and wifi)
- go to iOS App Store > no connection
- disable wifi > App Store does work
- enable wifi again > App Store still works

no blocks within query log or somewhere else.

Really annoying issue!

**Comment by hagezi** [comment]
@pictosun I have carried out the steps to reproduce the problem several times and cannot reproduce it.

**Comment by pictosun** [comment]
That really is a strange thing. Don't know what else I can do to troubleshoot and/or to find out, what's going on there?

**Comment by emeritaacuity0u** [comment]
> @hagezi 
> 
> After some testing I can say, that excluding fritz.box doesn't solve the issue.
> 
> 
> 
> Today I also recognized, that it also appears when coming out of flight mode). Maybe because disabling flight mode causes a short connection to LTE/5G and after that jumps to WIFI.
> 
> 
> 
> So to reproduce it:
> 
> - toggle flight mode (getting mobile network and wifi)
> 
> - go to iOS App Store > no connection
> 
> - disable wifi > App Store does work
> 
> - enable wifi again > App Store still works
> 
> 
> 
> no blocks within query log or somewhere else.
> 
> 
> 
> Really annoying issue!

These steps reproduce this issue for me also.

**Comment by pictosun** [comment]
Thanks for your feedback. So I'm not alone. Don't know what to do now.

Maybe the developers will give a hint how to check something else or maybe there is somewhere a bug from AdGuard DNS itself?!

Also found out, that there is no update over here since a longer time concerning the code...

**Comment by pictosun** [comment]
Today I found out, that the issue also exists, when using my WIFI-only iPad within my local WIFI network (and not changing the network). Just had it lying around and then I searched for updates within the iPad OS App Store and it says no connection.

I'm using the same Server (profile) within AdGuard DNS settings. So I checked again the logs and found, that I had a block from 'ca.iadsdk.apple.com' from a list (Hagezi) several time ago.

During the 'outage' of the App Store service (Apple Music was working), I tried it several times (and did not disable/enable WIFI > as this would solve the issue).
I checked the settings within the Server (profile) and changed my TTL from 3600 to zero.
And after this change the iPad OS App Store started directly. 

I checked again the logs and got one more block this time. It was quite close to the other one but little bit different 'tr.iadsdk.apple.com'. 
But even when blocking this request it works.

So maybe it must be something with setting up TTL and having requests to some domain and during the TTL blocking time apple changing the subdomain of the already blocked domain?
Scenario:
- App Store says no Internet connection
- logs say: ca.iadsdk.apple.com > blocked (a few minutes ago)
- changing TTL from 3600 to 0
- App Store works
- logs say: tr.iadsdk.apple.com > blocked
- but system still working

@hagezi Hope this is now a better scenario to find out what's going on there?!
I setup a TTL of 3600 for my mobile devices AND also a TTL of 3600 for my router (fritz.box DoT profile).
Router and devices nearly have the same lists.

And SORRY for saying that it doesn't show up within my blocking lists (but I never thought, that those iadsdk... domains can have an issue and I also didn't see them because of the TTL and having them little bit more down within my query lists.

@emeritaacuity0u can you check, if you also have setup a TTL for your servers/profiles?  
And maybe you can troubleshoot the way I did?

**Comment by hagezi** [comment]
I also use a block TTL of 3600, I cannot reproduce the problems with it. I can't imagine that this is the cause either.

**Comment by pictosun** [comment]
Ok - you're using the TTL in both variants? For your router and your clients? Or only within one Server (profile)?

**Comment by emeritaacuity0u** [comment]
![image](https://github.com/AdguardTeam/AdGuardDNS/assets/171407720/bd34b7f5-3822-4601-8887-e12ed49c5d32)
![image](https://github.com/AdguardTeam/AdGuardDNS/assets/171407720/d33f8aec-7ca9-4d43-9d48-16f49dff2537)
![image](https://github.com/AdguardTeam/AdGuardDNS/assets/171407720/e31b450d-0e6a-4b3c-8b6a-62529332dcb5)
![image](https://github.com/AdguardTeam/AdGuardDNS/assets/171407720/eaa9ef19-5dca-4e65-8f87-a2337abf99f0)

This is my setup. There's no other modifications.

**Comment by pictosun** [comment]
@emeritaacuity0u Did you try to set TTL to 0?

**Comment by pictosun** [comment]
As @hagezi said, the TTL was not the point. The issues are still there and happening again and again.

So as my iPad only has WIFI access I can say:
- changing TTL doesn't help
- blocklist also seems not to be the point
- mobile network vs. WIFI jumping through WANs is also not the point (iPad is WIFI only)

I'm a bit loss what it could be else if not some kind of implementation from AdGuard DNS itself (doesn't happen with other providers as already mentioned)

Maybe someone else has an idea how to troubleshoot this issue?

**Comment by hagezi** [comment]
I can't think of anything else at the moment. If you have blocked Private Relay, you could try to see if the problem still occurs when PR is not blocked.

**Comment by pictosun** [comment]
No - private relay is not blocked and I do have additional whitelisting of those domains.

I unticked the block private relay thing within AdGuardDNS and these are the additional whitelist rules:
- doh.dns.apple.com
- mask.icloud.com
- mask-api.icloud.com
- mask-h2.icloud.com
- doh.dns.apple.com.v.aaplimg.com

Today it already happened two times with my MacBook which is within my home WIFI only.

But it looks like it is more and more Apple App Store only. Sometimes it also happens with Apple Music (very rare). But must of the time it is the App Store on all devices.

<img width="509" alt="SCR-20240610-gqmd" src="https://github.com/AdguardTeam/AdGuardDNS/assets/55917962/dd83e5a1-939a-4f68-a0fc-770839671b23">

**Comment by hagezi** [comment]
I'm at the end of my tether too. I absolutely can't reproduce it, no-one in my immediate and extended family can. On none of the Apple devices and these are some that use AdGuard DNS via mobileconfig.

**Comment by pictosun** [comment]
As we can see that also @emeritaacuity0u has those issues I'm not alone at all. 

Maybe some developers like @Chinaski1 can jump in and help? Are you working on AdGuard DNS and maybe have looked into those issues? Will we see some changes coming with a new release in future?

@hagezi Concerning '*/cf/tr.iadsdk.apple.com' I do get a blocked result with NOERROR as answers for A,AAAA,HTTPS (it's always 3 requests for apple devices)
For 'mask.icloud.com' I do get unblocked and also NOERROR as answers.

I also installed LitteSnitch and tried it via DoQ profile on macOS (LS does use DNS Proxy network extension) and it happens even there. 
So it's not the DoH mobileconfig alone...

**Comment by hagezi** [comment]
Test the following on the iPad: Do not use the Fritzbox as the system DNS, but Cloudflare 1.1.1.1, for example, and continue to use the AdGuard DNS mobileconfig as the encrypted DNS.

See if this also occurs with a public DNS, which is then used as a bootstrap for the steering.

Apart from that, I can't really think of anything else you could try.

**Comment by pictosun** [comment]
> Test the following on the iPad: Do not use the Fritzbox as the system DNS, but Cloudflare 1.1.1.1, for example, and continue to use the AdGuard DNS mobileconfig as the encrypted DNS.

I don't get exactly what you're talking about...
If I setup a mobileconfig it does use the DNS over the mobileconfig. 

You mean leaving the mobileconfig enabled and change WIFI network DNS config from automatic to 1.1.1.1 for example? Or what do you mean?

**Comment by hagezi** [comment]
Yes, exactly that. The "WiFi DNS" is used as a bootstrap DNS to resolve the DNS domain from the moileconfig.

**Comment by pictosun** [comment]
@hagezi It looks like the test does work. When setting for example 1.1.1.1 as DNS within network settings the issue is gone (at least so far). 

So now it's getting interesting why this does happen and what's the cause?

**Comment by hagezi** [comment]
Next, use 1.1.1.1 as upstream in the FritzBox and the FritzBox as ‘WiFi DNS’ as before.

**Comment by pictosun** [comment]
I changed the DNS of fritz.box to 1.1.1.1 (one.one.one.one DoT within) and on all my devices back to standard (away from 1.1.1.1 back to automatic) and the error just came back on my iPhone in this case. (iPad is doing well - but I just started testing).

**Comment by Chinaski1** [comment]
@pictosun 
Hello there!

We're monitoring the situation but unfortunately we can't reproduce the issue on our end.

**Comment by pictosun** [comment]
> We're monitoring the situation but unfortunately we can't reproduce the issue on our end.

Thanks for your feedback. What I can say for sure is, that it's not the mobileconfig itself because it also happens on systems with dnsproxy installed and also when using DoQ instead of DoH.

And it also looks like it is not some blocklist itself. Must be something different.

Would be nice to get some feedback from @emeritaacuity0u as he is also having those issues.

**Comment by emeritaacuity0u** [comment]
I don't have anything more to add at this stage.
When using any AdGuard DNS server, my iOS App Store has connection issues intermittently. When I use another companies DNS server, it works.

**Comment by pictosun** [comment]
@emeritaacuity0u which Blocklists do you use?

**Comment by emeritaacuity0u** [comment]
AdGuard DNS filter.

The issue isn't happening with ControlD, or NextDNS for me.

**Comment by pictosun** [comment]
So it's definitively not the filter list at all. We do use completely different lists.

@Chinaski1 Maybe any advice what we both can do to troubleshoot more? What about the SERVFAIL fallback to CF and Google - could this maybe causing the issue for users?

**Comment by pictosun** [comment]
Could it be maybe something with EDNS (ECS ) implementation. When changing WAN networks (IPv4/IPv6) and maybe jumping to another EDNS (ECS) that the requests to some service (Apple App Store etc.) are somehow corrupted and you don't get correct responses?

Just an idea.

**Comment by ameshkov** [comment]
Let's try at least narrow it down to either filtering issues or the DNS server issues.

Please try using AdGuard DNS Unfiltered configuration (`unfiltered.adguard-dns.com`), if the issue can be reproduced with it then the culprit is filtering (filter list, access settings, etc, there are many things that change the normal DNS resolution).

If it's reproduced with AdGuard DNS Unfiltered, then we'll be looking from that side.

And one more thing. I guess we're at a point when to troubleshoot we need to look at the device logs.

**Comment by pictosun** [comment]
Hi @ameshkov 

I will try it. You mean generally using unfiltered.... as DNS mobileconfig for the devices and the router?

What about device logs? Which logs do you mean in general so that I can investigate.

**Comment by pictosun** [comment]
Just a short update. Today it started to get really bad again. As it is not always the same I do believe that it is definitely something ad AdGuard side, what is bringing up those issues.

@ameshkov What could it be, that the last days it was quite ok and with this (morning) it started to behave really bad and I'm getting more issues than the days before?!

**Comment by ameshkov** [comment]
> I will try it. You mean generally using unfiltered.... as DNS mobileconfig for the devices and the router?

The same way you setup the normal DNS.

> What about device logs? Which logs do you mean in general so that I can investigate.

If it's a macOS device there's a "Console.app" that has all the device logs. It should have the necessary info.

**Comment by pictosun** [comment]
> If it's a macOS device there's a "Console.app" that has all the device logs. It should have the necessary info.

@ameshkov Today I had those issues on my macOS machine and started console. 
Did some filtering of all the entries and could find, that there are some issues before the connection got working again. Don't know if any of those correlate to the issue but here are some:

- mash.hash does have some kind of issues and so on (CNAME) and answers like `mDNSCoreReceiveCacheCheck: rescuing RR with new TTL 15` and some `mDNSCoreReceiveNoUnicastAnswers: Removing expired record<mask.hash: ...>`
- an a lot of`AMSURLRequest: [...] Failed to fetch client ID domains from bag. Defaulting to not including analytics cookies. error = { Error domain=AMSErrorDomain, code=204 | URL = https://amp-api.apps.apple.com/v1/catalog/de/apps?REDACTED` before it will start running again
- when it starts working again I get: `AMSURLSession: [....] Task finished loading for URL request <AMSURLRequest: ....> { URL: https://amp-api.apps.apple.com/v1/catalog/de/apps?REDACTED }`
- `activating connection: mach=false listener=false peer=true name=com.apple.ak.anisette.xpc.peer`
- `failed to do a bootstrap look-up: xpc_error=`
- `ServerDataCacheService Finished running background update for ... ["deny-list"] failed`

Hope some of the messages can help to troubleshoot.
Unfortunately I only have very limited knowledge, but I would still like to make a guess...:
Could it be that it is because macOS performs/expects a certain caching here and if you open the App Store, for example, and the same server does not respond as stored in the caching (due to the dynamic change of the AdGuard DNS servers), that problems then occur as described? See #790

**Comment by emeritaacuity0u** [comment]
7590AX on FRITZ!OS 7.81.
I wont be able to try it out anytime soon.

As the issue happens both on and off the home network though, the router is not my single point of failure.

**Comment by pictosun** [comment]
Thanks for your feedback. 

So it must be something else what is causing those issues.

**Comment by pictosun** [comment]
@hagezi After some reading I tried another approach... I read somewhere, that errors can occur, when using a DoT profile with a number at the 'beginning' of the URL. So I tried to make a new profile (mobileconfig) starting with a letter. And will check, if this will work better? Don't know if this is true (I can't really imagine) - but I will give it a try.

Another point is maybe what I asked within #792 - and forgive me if I'm wrong with this. I'm not an expert - I try to help with my means as best I can.

**Comment by pictosun** [comment]
> And will check, if this will work better? Don't know if this is true (I can't really imagine) - but I will give it a try.

Doesn't help. The issue occurred again and again. 

So sad to see. Today I made some research and found out, that @emeritaacuity0u and myself are not alone. But they're not reporting it to Github.
Don't know, if they do switch to another provider or just ignore the issues.

Of course I'm now trying to click Apples App Store more often, than before. But it really is annoying to see all those outages while using AdGuard DNS.

And as I already said before it must be something on AdGuards end, as it really never happens with any other DNS provider (ControlD, NextDNS, native provider, Cloudflare, dnsforge and many others).

It really looks like AdGuard is doing something different in some way but I don't know what and why this makes it so unreliable (at my side).

**Comment by ameshkov** [comment]
> And as I already said before it must be something on AdGuards end, as it really never happens with any other DNS provider (ControlD, NextDNS, native provider, Cloudflare, dnsforge and many others).

Do you set them all up using a DNS profile?

**Comment by pictosun** [comment]
> Do you set them all up using a DNS profile?
> 
> In general, is it true that the problem only happens when you set up AdGuard DNS using a `.mobileconfig` file?

@ameshkov Thanks for your feedback. 
In earlier times I also tested DoQ and DoT for macOS (with Little Snitch 6 and it's new DNS proxy integration) but cannot remember if it also happens there or not as the macOS systems don't move that often to other WAN networks.

Concerning the mobileconfig:
- yes I do use mobileconfigs the last weeks and I cannot say if it also happens any other way as it's mobileconfig-only for me.

But when checking your .mobileconfig created from the dashboard it looks like I found a few small bugs.
- See the screenshots. It looks like there are some wrong characters within mobileconfig-creation.

Don't know if this is causing the (or any) issues? 
Overall it looks like a bug which should be corrected in general.
![mobileconfig_01](https://github.com/AdguardTeam/AdGuardDNS/assets/55917962/3323e539-d879-4984-9721-dd49726284b6)
![mobileconfig_02](https://github.com/AdguardTeam/AdGuardDNS/assets/55917962/0dc47ab3-3bd5-4f26-b433-1c46b3dceb01)

**Comment by pictosun** [comment]
@ameshkov Concerning the bugs within mobileconfigs should I open a separate bug issue or is it ok, when leaving it over here as you will solve the issue internally?

**Comment by ameshkov** [comment]
What do you use to view the internals of a signed mobileconfig?

**Comment by pictosun** [comment]
> What do you use to view the internals of a signed mobileconfig?

I'm using some kind of editors (CotEditor, BBEdit and so on). The important part of those files is always readable. 

If I open it with CotEditor in this case it shows:
![01](https://github.com/AdguardTeam/AdGuardDNS/assets/55917962/eddd01d0-7bee-49d6-a21c-7264b0846389)
![02](https://github.com/AdguardTeam/AdGuardDNS/assets/55917962/5d142f5c-ef7e-4774-8778-26ce307ecff7)
![03](https://github.com/AdguardTeam/AdGuardDNS/assets/55917962/8e46c3d2-2897-4dd7-844d-25b38a5f967c)

So different but also not regular character set.

And as I already opened/edited and worked with many .mobileconfigs and never have seen such inconsistencies within this part of this configs I thought, that this could be a bug at all.

**Comment by ameshkov** [comment]
My point is that with the errors that the editor shows macOS/iOS wouldn't be able to import the file so most likely the problem is with the editor.

**Comment by pictosun** [comment]
> Anyways, have you tried a mobileconfig with server addresses specified? Does it help?

Cannot say for sure. I'm still testing. One config with IPs and the other one with hostnames - both are running at the moment. The problem doesn't exist all the time - looks like it sometimes happens more often than other times... 

So it's not that easy to find out what's the issue at all.

**Comment by pictosun** [comment]
Just to give a feedback, that it also happens on systems using DoQ and DNSproxy (like LittleSnitch is offering). So it's not a mobileconfig issue at all.

Happened a few minutes ago again... After a special amount of time it starts working again and the first entries within the logs are requests to CDNs (but that's normal I think because of Apple using CDNs for their App Store). 
But maybe that's something to put in consideration when finding out the issue?!

**Comment by pictosun** [comment]
And another short update... Next device is also having issues. I think it could be something correlated to #795 as I'm seeing huge spikes there when checking TCPPing, DNS probes etc.

**Comment by pictosun** [comment]
And the next device is getting those errors... At the moment a lot of issues.

**Comment by pictosun** [comment]
@emeritaacuity0u Any update(s) from your side?

**Comment by emeritaacuity0u** [comment]
No, I do not have time to test.

**Comment by ameshkov** [comment]
Well we tried every indirect way to figure out what's happening and I guess at this point the only thing that can answer this question is the device logs exported from console app + the exact time when the issue happened + a detailed explanation of what was happening at that time (how did you understand that something is wrong and what happened next).

**Comment by pictosun** [comment]
Today a lot of issues appeared again.

As mentioned a few posts earlier it could really be something because of the errors within automatic .mobileconfig configuration profiles when downloading them via dashboard.

See the error message within console.

<img width="615" alt="SCR-20240714-rblh" src="https://github.com/user-attachments/assets/fd76ea90-a121-4217-98c4-6aefe892cc42">

**Comment by pictosun** [comment]
@ameshkov Enclosed some more details to my latest post. It really looks like the issue is coming from your side (somehow).

`NESMDNSSettingsSession[d.adguard-dns.com DoH:REDACTED]: Removing a connection for client Network[REDACTED]

NESMDNSSettingsSession[d.adguard-dns.com DoH:REDACTED]: Removing a connection for client VPN[REDACTED]

AMSTreatmentStore: [REDACTED] Failed to fetch areas (error: { Error domain=NSCocoaErrorDomain, code=260)

TCP Conn [23:REDACTED] using empty proxy configuration

activating connection: mach=false listener=false peer=false name=(anonymous)

in_progress parent-flow (satisfied (Path is satisfied), interface: en5, ipv4, ipv6, dns)

nw_protocol_boringssl_signal_connected

nw_endpoint_resolver_update [C5.1.1 Hostname#REDACTED:443 in_progress resolver (satisfied (Path is satisfied), interface: en5, ipv4, ipv6, dns)] Adding endpoint handler for IPv6#REDACTED.443`

**Comment by ameshkov** [comment]
A couple of lines without other details don't help much, sorry.

We need three things to troubleshoot, not having one of three makes troubleshooting impossible.

1. Issue description (what happened, what error have you seen in the UI)
2. Exact time of the issue
3. Logs exported from console app (hand-picked lines are not enough since we don't know what we're looking for yet). I understand that sharing the whole log file might be too much, but can you at least export a 2-3 minutes period that corresponds to the time when you experienced the issue?

**Comment by pictosun** [comment]
Thanks for your feedback.

The point is, because of privacy reasons I cannot send you those logs as it's too much private information inside.

What's happening all the time is that I cannot connect to Apple services when using your DNS services within my setup. (see screenshot).
When reconnecting to the network(s) it does work again (don't know if it's a caching issue or something else?). 
And it does only happen when using AdGuard DNS services.... 

Left is Apple Music, right is Apple App Store

I have an idea:
I think I will try now a .mobileconfig using the same DoH setup as your generated .mobileconfig one and doublecheck if it is working without any issues... (will use one generated by https://dns.notjakob.com)
But overall I don't think it will solve the issues as I already tried it via DNSproxy and LittleSnitch DNS configurations.

But let's see if I see any difference.

<img width="2403" alt="SCR-20240714-qvkk" src="https://github.com/user-attachments/assets/2507579b-9c94-45d2-93e1-4b09641d6f27">

**Comment by Wrestor** [comment]
I don't know if this is mentioned in this thread before, but this issue affects every platform, the AdGuard dns profile is installed to (iOS, iPadOS, macOS & visionOS). On my end, even though I excluded my WiFi SSID, this bug still affects me!

Note: I have always had my router set to the public AdGuard dns (ipv4 & 6), it isn't affected by this bug though.

**Comment by pictosun** [comment]
@Wrestor Did you install the .mobileconfig from AdGuard DNS Dashboard or created one by yourself? Does the issue happen all the time at your side (even if it is difficult to control it all the time).

**Comment by Wrestor** [comment]
@pictosun I'm using the Default Server configured via "[Method No2](https://adguard-dns.io/en/public-dns.html): Configure AdGuard DNS manually" > iOS > open profile constructor.

**Comment by pictosun** [comment]
@Wrestor Thanks for the feedback.

Is it possible for you to try out some stuff. 
For me I'm testing an self created mobileconfig and since yesterday it does work well without any issues (but it's only a small amount of time).
You can create them for yourself via the entry above from @ameshkov or for example via this one https://imazing.com/profile-editor or https://dns.notjakob.com

Will test a bit longer and when everything is doing will it looks like the AdGuard DNS dashboard profile creator does have some issues (but at the moment it's to early to say).

Maybe you can also jump and test so that I'm not alone here.

**Comment by emeritaacuity0u** [comment]
Unfortunately I've had to do the same. The issue isn't isolated to the mobile config's and happens when you use the AdGuard for iOS app with either native or AdGuard DNS implementation.

I've used DoT, DoH2, and DoH3 with AdGuard Personal DNS with the same outcome.

**Comment by pictosun** [comment]
So let's hope someone somehow finds out what those issues are... @emeritaacuity0u @Wrestor are you still 'on board' or already jumped to another provider?

**Comment by pictosun** [comment]
@hagezi and/or @ameshkov Could it be also be some weird setting within iCloud Private Relay, Hiding IP Addresses from Trackers, Prevent use advanced tracking and fingerprinting protection or something else?

I've seen within your knowledge base at the website, that you describe a few settings there, but as far as I know @hagezi is using some of the settings?

Gerd, how did you manage those settings in detail?
- iCloud Private Relay on/off?
- separate network settings for hiding IP Addresses on/off?
- advanced tracking and fingerprinting protection within Safari on/off?
- Settings within Safari and mail concerning hiding IP addresses from trackers on/off?

And overall en- or disabled the setting "Block access to iCloud Private Relay" within AdGuard-DNS Dashboard?

Maybe that's the reason of all those issues?

**Comment by ameshkov** [comment]
I am not entirely sure if iCloud Private Relay is used by iTunes app tbh. Do you have it enabled?

**Comment by pictosun** [comment]
No - I don't have it enabled, but there many other settings which could also be some kind of influencing iTunes and especially Mac/iOS App Store where the issue occurs...

Maybe @hagezi can jump in an say how he has configured all those settings as mentioned above. So it's easier to troubleshoot from my side as there are too many combinations to test.

**Comment by hagezi** [comment]
You have written that you have disabled these options, so it cannot be the cause of your problem. For me, it also makes no sense to play the combination lottery with these options.

Just do the test that has already been requested here, take the unfiltered AdGuard DNS as mobileconfig and see if the problem also occurs with it, see https://github.com/AdguardTeam/AdGuardDNS/issues/777#issuecomment-2180172987

Furthermore, according to one of your statements, it works if you use Cloudflare 1.1.1.1 as the system DNS instead of your router, for example, and at the same time a mobileconfig that uses an AdGuard DNS profile. The steering (the resolution of the name from the Mobileconfig) then takes place via the Cloudflare DNS and not via your router and the DNS configured there. You could then search further in this area, but everything has already been said.

**Comment by pictosun** [comment]
> Just do the test that has already been requested here, take the unfiltered AdGuard DNS as mobileconfig and see if the problem also occurs with it, see [#777 (comment)](https://github.com/AdguardTeam/AdGuardDNS/issues/777#issuecomment-2180172987)

Using whatever inside fritz.box and take #777 comment as mobileconfig within the iOS device you mean?

**Comment by dowered-baize** [comment]
I'm seeing this occur also although I have only seen it occur with the App Store on iOS (17.6)and macOS (14.6). I have seen it occur with both the mobileconfig profile, the AdGuard iOS app, and plain dedicated IPv4 that I have through AdGuard DNS. It happens most frequently on iOS, and it seems that changing from cellular to wifi (or vice-versa) causes it to happen - although as already mentioned in this thread, it does seem to occur randomly, and therefore is difficult to reproduce. I am only seeing this occur with AdGuard DNS, and not ControlD using the same lists (HaGeZi's Pro++, HaGeZi's Threat Intelligence Feeds). I'm in the US and currently connected to the NYC AdGuard DNS server.

**Comment by Wrestor** [comment]
I am now also running into another issue. Since iOS 17.6, the AdGuard DNS Profile seems to be (visually?) missing its certificate.
<img src="https://github.com/user-attachments/assets/8864f13a-5417-40ff-975e-08787de72709" width=25% height=25%>
<img src="https://github.com/user-attachments/assets/00cd1ff1-9f20-4cb5-89b1-ebaa091258be" width=25% height=25%>

**Comment by pictosun** [comment]
Any update over here, as it looks like, I'm not the only one having this issue?

**Comment by pictosun** [comment]
> I am now also running into another issue. Since iOS 17.6, the AdGuard DNS Profile seems to be (visually?) missing its certificate.

For me it looks the same as with other providers.

**Comment by pictosun** [comment]
> I'm seeing this occur also although I have only seen it occur with the App Store on iOS (17.6) and macOS (14.6). 
@dowered-baize  Does it still happen at your side? It really looks like there is some error from AdGuard as reports are getting more and more.

Good to see, that I'm not the only one. Hope they will solve those issues in near future.

**Comment by pictosun** [comment]
> Unfortunately I've had to do the same. The issue isn't isolated to the mobile config's and happens when you use the AdGuard for iOS app with either native or AdGuard DNS implementation.
> 
> I've used DoT, DoH2, and DoH3 with AdGuard Personal DNS with the same outcome.

Any update from your side? Are you still seeing those issues?

**Comment by emeritaacuity0u** [comment]
I've got 2 devices still using AdGuard DNS. One's on iOS beta, one's on iOS stable. Both still experience this issue.
The remainder of my devices are using different upstream with no App Store issues.

I also have an issue with unfiltered AdGuard DNS with the Binge streaming service. After about 20mins of using it, it locks up the entire TV, preventing it from even turning off.
This behaviour isn't seen with Cloudflare DNS or Google DNS.

**Comment by pictosun** [comment]
@emeritaacuity0u  
> I've got 2 devices still using AdGuard DNS. One's on iOS beta, one's on iOS stable. Both still experience this issue. The remainder of my devices are using different upstream with no App Store issues.

That really is the part what I cannot understand? Why do we have issues, while others don't... Would love to see the issues solved as it's nearly unusable for me/family when having those issues.

**Comment by pictosun** [comment]
@ameshkov Any update from AdGuard here? As I'm not alone and issues reoccur when using AdGuard DNS and don't exist when using other providers it must be something wrong with AdGuard. Maybe @emeritaacuity0u  can also jump in and give some details.

**Comment by pictosun** [comment]
1.5 months later and still no response. 

The error is still there and I cannot get it away. So sad that this issue is still existing (and as you can see I'm not the only one). Why on earth does it only happen with AdGuard and not with ControlD, NextDNS and other providers? You must be doing something different...

**Comment by hagezi** [comment]
1.5 months later and you still haven't delivered what you're supposed to. What do you think will happen here without you providing the required details if this is not reproducible for 99.5% of users?

Unfortunately, the crystal ball has no solution.

**Comment by pictosun** [comment]
What I found out today is, that within AdGuard DNS logs "h3.apis.apple.map.fastly.net" appears - it starts to work again.

Concering 1.5 months and nothing delivered. I showed that automatic mobileconfig does have bugs/errors within it's files and my console log also shows that the mobileconfig has errors. Maybe it has something to do with it or not - I really try to help as good as I can - but I cannot send too much private details.

**Comment by hagezi** [comment]
> Concering 1.5 months and nothing delivered.

How could they? There are no details to narrow down the error. Logs ...

As far as the mobileconfig is concerned, it is valid and works. [Here you were asked to provide the profile](https://github.com/AdguardTeam/AdGuardDNS/issues/807#issuecomment-2295956335) for further investigation, did you do that?

**Comment by emeritaacuity0u** [comment]
> As far as it is possible for me to send infos - yes. But I cannot send the whole profile. 
> 
> 
> 
> Can you maybe check (have a look into your generated mobileconfigs and check if you do have the same issues within those files as I do have.

Why can't you send the whole profile?

**Comment by hagezi** [comment]
> As far as it is possible for me to send infos - yes. But I cannot send the whole profile.

Create a new device, generate the profile, make it available for further investigation, delete the device ...

**Comment by pictosun** [comment]
> Create a new device, generate the profile, make it available for further investigation, delete the device ...

I know - but there is too much personal data within. 
In future I will built a test scenario with other machines, other networks and separate user-account - so to be sure that no personal/private data will be transmitted.

In the meantime you can also look at your mobileconfigs. 
I'm pretty sure they do show up the same glitches/issues (as long as you create them via AdGuard DNS Web GUI).

Within console.log from macOS then the errors will pop up.

And don't get me wrong, I'm trying to help as much as I can... And to simply say all the time give us more data, and then we will look into it.
I already gave so many hints/stats/infos... Overall I'm a paying subscriber and just want to have a service which does work without any issues.

I try to help as much as I can/will.

**Comment by Critical-Break-5818** [comment]
Hey guys, i also have same problem. Appstore not loading. I am paid user of Adguard DNS. Using Hagezi recommended blocklist only. Not using Adguard dns filter list.

I have connected to Adguard DNS using profile only. All of my apple devices ios, macos, tvos, ipados facing same issue that appstore not loading. Please help to resolve the issue

**Comment by pictosun** [comment]
@Critical-Break-5818 I'm having those issues for a long time and it does still exist. On iOS systems I can simply disable/enable WIFI/Ethernet connection and the issue is solved. Could you please try if it is the same for you?
Otherwise I still don't know what is causing those issues. The only thing I can say is that it only happens with AdGuard DNS. Others like NextDNS/ControlD and so on don't have those issues.

**Comment by Critical-Break-5818** [comment]
Now i have created Test server on Adguard DNS and moved my ios devices on that server. Also disable all settings like blocklist, access settings, security and parental control. Lets see how it goes. Will keep you posted

**Comment by Critical-Break-5818** [comment]
@pictosun @ameshkov Not i am not using any blocklist, access setting or parental or security. ALL ARE DISABLED

I am connected to Adguard DNS through profile created by Adguard itself. 

Still Apple Appstore not working.
Attached photo
![IMG_0823](https://github.com/user-attachments/assets/0a14a678-4e91-4a4c-b82e-71368f7df465)

**Comment by Critical-Break-5818** [comment]
Yes you are right. Toggling wifi reset the connection and able to connect with Appstore again. 

There is some issue with networking. Something reset after toggling and able to connect again with AppStore.

**Comment by MineGene** [comment]
Curiosities sake, have you had Private Relay turned on or off for the testing?

Per Apple's documentation, profiles and Private Relay are meant to work together without issue, but it couldn't hurt to try with it turned off if it was on.

**Comment by pictosun** [comment]
> Curiosities sake, have you had Private Relay turned on or off for the testing?
As I'm having the same issue since started using AdGuard DNS I nearly tried everything in my hands to solve this issue (also turning on/off Private Relay, changing devices, trying different configs/profiles and so on.

**Comment by Critical-Break-5818** [comment]
Private relay is turned OFF.. 

Also one thing that i have noticed is that,  Under WIFI settings, i changed PRIVATE WIFI ADDRESS TO FIXED INSTEAD OF ROTATING. Now issue is less frequent. 

Try changing this setting then me know

**Comment by Critical-Break-5818** [comment]
First change the settings to OFF, Save it , turn off wifi then TURN it ON then finallychange it to FIXED, and complete shutdown then phone for 30 sec then restart it.

this is the hack. Pls try it

**Comment by pictosun** [comment]
> this is the hack. Pls try it

Tried it several times with several devices (long time ago already) but it doesn't help. In my home (well known) networks I always have fixed WIFI (real MAC-ID) address activated. And also tried the other settings.
I think the issue is not related to the devices I'm (and other family members) are using.

**Comment by pictosun** [comment]
@Critical-Break-5818 Any update from your side?

**Comment by Critical-Break-5818** [comment]
Issue is not with filter thats for sure. Its within Adguard DNS. 

I have changed WIFI to Fixed. So far, i am facing the issue one or two times in a day per device.

**Comment by pictosun** [comment]
Ok. But I think, that WIFI fixed or not will make any difference. I think it has something to do with the implementation from AdGuard at all. Some times it is much more often than in other times.
For me I can see some correlation between their DNS server utilization and the issue appearing. 

I do track different DNS providers via Smokeping and can see that AdGuard does have a lot of 'smoke' when looking at smokeping results and comparing to other providers. (In peak times - during daytime in Europe) the issue occurs much more.

Maybe I'm wrong, but that's the only thing I found out so far.

**Comment by Critical-Break-5818** [comment]
@ameshkov Please help.

**Comment by pictosun** [comment]
@Critical-Break-5818 Did you made some improvements or got some feedback concerning the issues? For me, it is always a spark of hope when I am not the only one with the problem and it simply doesn't go away.

I hope that the issue can be resolved someday. It doesn't seem to be my setup, as I am not the only one with the problem.

**Comment by Critical-Break-5818** [comment]
Issue seems to be not happening now. I haven’t changed anything from my side.

**Comment by pictosun** [comment]
For me it's also getting better (no issues). But loading the store takes longer than with other providers.

@ameshkov Could it maybe something that is correlated to the request / response speed of your DNS servers? If it takes to long, than maybe the Store is showing the status as said above?

For example when looking at this ncie test site:
https://github.com/thomasmerz/dnspingtest_rrd_ka
And showing his comparison:
https://thomasmerz.github.io/dnspingtest_rrd_ka/dashboard_week.html

You can see in detail for AdGuard (the other AdGuard servers he is monitoring showing the same), that especially in daytime (EU) there are a lot of huge peaks within AdGuard
https://thomasmerz.github.io/dnspingtest_rrd_ka/index_94.140.14.14_adguard_mit_werbe_und_trackingfilter.html

Why are there those huge peaks and with other providers we don't have those peaks?

**Comment by Critical-Break-5818** [comment]
Appstore is loading but very slow.

**Comment by pictosun** [comment]
Back at home I found out, that my Smokeping setup from my Homeserver showed up some issues with AdGuard DNS exactly at the time I had the issues.

Setup is a Smokeping Server with AnotherDNS testing - https://oss.oetiker.ch/smokeping/probe/AnotherDNS.en.html

See the screenshot.

<img width="707" alt="Image" src="https://github.com/user-attachments/assets/99749e27-3027-48f9-8d59-07fa517b7c43" />

@ameshkov Maybe this a hint concerning all the issues?!

**Comment by pictosun** [comment]
@Critical-Break-5818 Are you using AS3320 (DTAG) as network?

**Comment by Critical-Break-5818** [comment]
What is that?

**Comment by Critical-Break-5818** [comment]
I am using TMobile USA

**Comment by pictosun** [comment]
Maybe it has something to do with routing to T-Mobile in general? @ameshkov Is this something you can check from your side?

**Comment by pictosun** [comment]
In my testing I found out, that AdGuard DNS servers have performance issues compared to other providers. I tested it with two complete seperate WAN networks (totally different ASNs) and separate testing scenarios.

Maybe this is the issue bringing the problems up with the App Store. 

If you're all able to test yourself. 
You can check it with DNS Probe, Another DNS and other testing from the excellent Smokeping Setup found here:
https://oss.oetiker.ch/smokeping/probe/index.en.html

It's not that hard to setup such a scenario and do some testing. Hope someone can also set it up so that we can compare.

During daytime in Europe the AdGuard DNS servers (tested two filtered and two unfiltered IPs) do have issues/smoke and variations and peaks up to 500-600ms. 

Others like Standard DNS, Quad9, ControlD, NextDNS, dnsforge.de and so on don't have those issues.

**Comment by pictosun** [comment]
@MineGene 

I installed Smokeping on different Proxmox machines via LXC (Debian).
Used the AnotherDNS, fping, DNS Probe and other tools to diagnose against several IPv4 addresses of different DNS providers.
And within a few days you can see, that only AdGuard DNS does have those issues. 
To rule out that it is just an issue with my connection, I set it up and tested it as a test scenario in two other environments as well and came to the same result.

**Comment by Monotone4964** [comment]
adguard support sent me to this thread since they had no idea why my appstore downloads are dog slowwwwwww...... Definitely an issue with Adguard DNS because like others if i change DNS speeds are fast then.

**Comment by pictosun** [comment]
Thanks for your feedback over here. I really think that it is because of the AdGuard DNS servers itself. In all my testing it shows capacity issues with AdGuard servers - I think that this is a valid point.

**Comment by pictosun** [comment]
@ameshkov 

Are you still working on this issue? It definitely looks like your servers have issues with fast answers to all those requests. Sad to see no response over here.

All others I asked also do have issues concerning this point and a few of them also started logging DNS requests/responses to your (and others) servers. AdGuard is the only one with those issues.

**Comment by pictosun** [comment]
Another update. Yesterday I found out, that a friend of mine is having exactly the same issues. He is using a separate account, another ISP, an other systems but is having the same issues (App store not responding and so on).

So for me it's clear, that this is not an issue at my/our side and has something to do with AG DNS servers or configuration at all?!

**Comment by pictosun** [comment]
Could it be possible that the Cloudflare / Google fallback mechanism play some role over here also?

Another thing I found out is, that when I adjust the TTL down to something like 10seconds (from 60 or more) it happens not that often then before (but this can also be a temporary situation).

**Comment by pictosun** [comment]
Short update here... Changing TTL doesn't help. Still having those outages.

**Comment by pictosun** [comment]
Any update over here? The last week I tested it in several countries and also widened the test scenario to a few friends. All having the same issue.

**Comment by pictosun** [comment]
@Chinaski1 @ameshkov @hagezi 

I just found this answer at Reddit (from AdGuard) concerning a similar issue. Could this also be related to the issue stated over here?
The answer from Ad_h3lper:
"Hello! This is a known issue which is caused by Apple's use of Akami as SDN and that Akami doesn't use EDNS Client Subnet. Unfortunately, as of now, this behavior is only caused by Apple and there's nothing we can do on our side. Whitelisting the domains mentioned in this post is a viable solution for now, but we hope it will eventually get fixed on Apple's side."

**Comment by hagezi** [comment]
As I have already mentioned several times, the problem cannot be reproduced for me, my family or my friends when using AdGuard DNS. 

Which domains to unblock are mentioned in the article? Doesn't make any sense in this context.

**Comment by pictosun** [comment]
> Which domains to unblock are mentioned in the article? Doesn't make any sense in this context.

https://www.reddit.com/r/Adguard/comments/1ehps9h/adguard_blocks_ios_app_store/

**Comment by hagezi** [comment]
How unblocking non-blocked domains is supposed to solve the problem makes no sense to me.

**Comment by pictosun** [comment]
I'm afraid I'm going to give up... 
No matter on which devices, with which accounts. AdGuard DNS keeps causing the same problems and there is no solution.
Other providers don't have these problems.
And I have tried various accounts (including friends), various devices, various WAN networks and countries...
Always the same problems as described above. 
And only with AdGuard DNS.
Maybe it's also because others don't look “so closely” or don't care.

**Comment by pictosun** [comment]
> How unblocking non-blocked domains is supposed to solve the problem makes no sense to me.
I still think it has something to do with this point... https://www.reddit.com/r/Adguard/comments/1ehps9h/adguard_blocks_ios_app_store/?sort=new

But for now I'm done with AdGuard DNS (sadly - as I really like the service). I will come back in maybe a year or so

**Comment by pictosun** [comment]
@m-rue Thanks for jumping in here... Since my last post I didn't try AdGuard DNS anymore as it was driving me (and other family members) crazy because of this App Store issues.

Using NextDNS/ControlD works without any issues. 

After seeing your other issue #792 with the note about whether/that this might solve the issue here, I tried AdGuard DNS again—but the errors mentioned came back immediately.
It is definitely something in AdGuard DNS that is causing the problems.

Hope to get this issue solved as it really is annoying.

**Comment by pictosun** [comment]
So annoying. And it definitely is some kind of bug from AdGuard DNS as others like ControlD don't have this issue.

Cannot understand that @hagezi who also uses AGDNS never had this kind of issues??!

**Comment by m-rue** [comment]
> [@m-rue](https://github.com/m-rue) What about your setup and maybe you can give some more details as Hagezi mentioned above?

I have a very similar setup. At home I'm using DoT in my router (FritzBox). On iPhone and iPad I'm using DoH with AdGuards .mobileconfig. Recently it has become very difficult to reproduce the issue. So far it happend only twice for me the last two weeks.

But I've just discovered that issue #451 seems to be fixed now. I'm getting now NXDOMAIN for every DNS record type. Maybe this also correlates with the issue here? Because now, iCloud Private Releay seems to be blocked correctly. I think we should check now, if it still happens.

**Comment by pictosun** [comment]
> Maybe this is a routing problem but I don't know. Which provider do you use? I'm using Telekom (T-Online).

Using Telekom. And it also does happen when routed via FRA (checked with dnscheck.tools). 

Maybe it can also be some kind of caching or routing issue as during the time of the error the devices will try to connect to some CDNs *.akamai.net *.akamaiedge.net *.akadns.net 

Maybe it is something correlated to the ECS as ECS is often seated somewhere else as those DNS servers....

It's really hard to tell.

@m-rue What did you setup within SERVERS > SERVER SETTINGS > ADVANCED > 'Respond to blocked domains'? For me I use 'Zero IP Address' as it is the default.

**Comment by pictosun** [comment]
As this issue is still happening at all of my/and family devices... Is there any update from other users? Is this happening for you too?
@m-rue what about you?

**Comment by m-rue** [comment]
> As this issue is still happening at all of my/and family devices... Is there any update from other users? Is this happening for you too? [@m-rue](https://github.com/m-rue) what about you?

Yes it is still happening, but not as often as before.

**Comment by pictosun** [comment]
> Yes it is still happening, but not as often as before.

Thanks for your feedback. Good to see, that I'm not alone with this issues (and I think many others don't care). If you don't know about it, then you might happen to see it sometime—but it's pretty unlikely as many do not force to manually update Apps. 
Maybe they will see it more often when using Apple Music.

I really like to know why this is happening - and not happening with ControlD or others... When askting AI it has something to do with how adguard-dns.io handles it under the hood as there are no blocked services or other things happening.

**Comment by matzilein** [comment]
Du bist nicht allein. Seit über 1 Jahr besteht das selbe Problem auch bei mir. Ob DoH oder DoT. Ob Mobilfunk oder Im Heimnetz über die FritzBox. Auf dem iPhone und auf dem IPad. Eingerichtet mit dem Konfig. Profil. In der FritzBox als DoT und auf dem iPhone auf automatisch … selbes Problem. Egal als Konfig.Profil oder DoT in der FritzBox. Es scheint ein Fehler in der Infrastruktur von AdGuard DNS zu sein. Und mir ist es ein Rätsel, wieso man dafür so lange benötigt um das Problem  zu beseitigen.
Inzwischen nutze ich parallel dnsforge. Wenn Adguard das Problem behoben hat wechsle ich gerne zurück. 

Grüße
